src/main/java/org/spongepowered/api/event/impl/AbstractModifierEvent.java

```diff
@@ -56,20 +56,20 @@
     protected ImmutableList<T> init(double originalValue, List<T> originalFunctions) {
         final ImmutableList.Builder<Tuple<M, Double>> modifierMapBuilder = ImmutableList.builder();
         final ImmutableList.Builder<T> functionListBuilder = ImmutableList.builder();
-        final ImmutableMap.Builder<M, Double> mapBuilder = ImmutableMap.builder();
+        final Map<M, Double> mapBuilder = new LinkedHashMap<>();
         double finalDamage = originalValue;
         for (T tuple : originalFunctions) {
             this.modifierFunctions.add(convertTuple(tuple.getModifier(), tuple.getFunction()));
             double tempDamage = checkNotNull(tuple.getFunction().applyAsDouble(finalDamage));
             finalDamage += tempDamage;
             modifierMapBuilder.add(new Tuple<>(tuple.getModifier(), tempDamage));
```

Comment:
Does this line need to be changed as well?

Reply:
No, they are lists